src/Constraint/MultiConstraint.php

```diff
@@ -111,7 +111,7 @@ public function matches(ConstraintInterface $provider)
     {
         if (false === $this->conjunctive) {
             foreach ($this->constraints as $constraint) {
-                if ($constraint->matches($provider)) {
+                if ($provider->matches($constraint)) {
```

Comment:
Not sure what the idea is here, but pretty sure you can't do this change safely.

Reply:
oh, constraint are not swappable? I thought `$a->matche($b) === $b->matches($a)`
The idea were a small optimization:
- `Constraint` is not able to matches `MultiConstraint` and fallback to the provider
- The probability that `$constraint` beiing a `Constraint` is higher than `$provider` beiing a `Constraint`

By swapping the call, it avoid `$constraint` to fallback to `$provider->matches($this)` and avoid 1 method call

If you are not sure about the assumption `$a->matche($b) === $b->matches($a)` I can revert

Reply:
Well, just have a look at the test suite result :)

Reply:
reverted